app/src/test/kotlin/fr/free/nrw/commons/nearby/CheckboxTriStatesTest.kt

```diff
@@ -0,0 +1,62 @@
+package fr.free.nrw.commons.nearby
+
+import android.widget.CompoundButton
+import androidx.test.core.app.ApplicationProvider
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.verifyNoMoreInteractions
+import fr.free.nrw.commons.TestCommonsApplication
+import fr.free.nrw.commons.location.LatLng
+import fr.free.nrw.commons.nearby.CheckBoxTriStates.CHECKED
+import fr.free.nrw.commons.nearby.CheckBoxTriStates.UNCHECKED
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [21], application = TestCommonsApplication::class)
+class CheckBoxTriStatesTest {
+    @Mock
+    internal lateinit var callback: CheckBoxTriStates.Callback
+    @Mock
+    internal lateinit var onCheckChangeListener: CompoundButton.OnCheckedChangeListener
+    private lateinit var checkBoxTriStates: CheckBoxTriStates
+
+    /**
+     * initial setup
+     */
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        MockitoAnnotations.initMocks(this)
+        checkBoxTriStates = CheckBoxTriStates(ApplicationProvider.getApplicationContext())
+        checkBoxTriStates.setCallback(callback)
+        checkBoxTriStates.setOnCheckedChangeListener(onCheckChangeListener)
+    }
+
+    @Test
+    fun testSetStateWhenSameState() {
```

Comment:
Hi @neslihanturan , could we get brief Javadocs for each of the tests please? Happy to merge when that is done. :)

Reply:
Done :)